packages/core/utils/src/__tests__/query-populate.test.ts

```diff
@@ -104,4 +104,75 @@ describe('traverseQueryPopulate', () => {
 
     expect(query).toEqual('address');
   });
+
+  test('should work with filters attribute', async () => {
+    // Expect should be called 4 times
+    expect.assertions(3);
+
+    const strapi = {
+      getModel: jest.fn((uid) => {
+        return {
+          uid,
+          attributes: {
+            filters: {
+              type: 'string',
+            },
+          },
+        };
+      }),
+      db: {
+        metadata: {
+          get: jest.fn(() => ({
+            columnToAttribute: {
+              address: 'address',
+            },
+          })),
+        },
+      },
+    } as any;
+
+    global.strapi = strapi;
+
+    const schema = {
+      kind: 'collectionType',
+      attributes: {
+        title: {
+          type: 'string',
+        },
+        address: {
+          type: 'relation',
+          relation: 'oneToOne',
+          target: 'api::address.address',
+        },
+      },
+    } as const;
+
+    const ctx = {
+      schema,
+      getModel: strapi.getModel,
+    } as const;
+
+    await traverseQueryPopulate(({ key, parent }) => {
+      switch (key) {
+        case 'address':
+          // top level populate should not have parent
+          expect(parent).toBeUndefined();
+          break;
+
+        case 'filters':
+          // Parent information should be available
+          expect(parent.key).toBe('address');
+          expect(parent.attribute).not.toBeUndefined();
+          break;
+        default:
+          break;
+      }
```

Comment:
Could we also add an expect on the attribute itself? To make sure both have the correct values?

```suggestion
    await traverseQueryPopulate(({ key, parent, attribute }) => {
      switch (key) {
        case 'address':
          // top level populate should not have parent
          expect(parent).toBeUndefined();
          expect(attribute).toBeDefined();
          break;

        case 'filters':
          // Parent information should be available
          expect(parent.key).toBe('address');
          expect(parent.attribute).not.toBeUndefined();
          expect(attribute).toBeDefined();
          break;
        default:
          break;
      }
```